src/tests/Microsoft.PowerFx.Core.Tests/ParseTests.cs

```diff
@@ -0,0 +1,719 @@
+
+        // Nested
+        [InlineData("{ A:{  }, B:{  }, C:{  } }")]
+        [InlineData("{ A:{ X:10, Y:true }, B:{ Z:\"Hello\" }, C:{ W:{  } } }")]
+        [InlineData("{ A:{ X:10, Y:true }, B:{ 'ZZZZZ':\"Hello\" }, C:{ 'WWW WWWW WWWW':{  } } }")]
+        public void TestParseRecords(string script, string expected = null)
+        {
+            TestRoundtrip(script, expected);
+        }
+
+        [Theory]
+            [InlineData("{{}}")]
+            [InlineData("{ , }")]
+            [InlineData("{A:1, }")]
+            [InlineData("{A:1,,, }")]
+            [InlineData("{A:1, B:2,,, }")]
+            [InlineData("{ . . . }")]
+            [InlineData("{ some identifiers }")]
+            [InlineData("{ {some identifiers} }")]
+            [InlineData("{{}, {}, {}}")]
+            [InlineData("{ 10 20 30 }")]
+            [InlineData("{10, 20, 30}")]
+            [InlineData("{A; B; C}")]
+            [InlineData("{A:1, B C}")]
+            [InlineData("{A, B, C}")]
+            [InlineData("{A B C}")]
+            [InlineData("{true, false, true, true, false}")]
+            [InlineData("{\"a\", \"b\", \"c\"}")]
+            [InlineData("{:, :, :}")]
+            [InlineData("{A:10; B:30; C:40}")]
+            [InlineData("{A:10, , , , C:30}")]
+            [InlineData("{{}:A}")]
+            [InlineData("{10:B, 20:C}")]
+            [InlineData("{A:10 B:20 C:30}")]
+            [InlineData("{A:10 . B:20 . C:30}")]
+            [InlineData("{A;10, B;20, C;30}")]
+            [InlineData("{A=20, B=30, C=40}")]
+            [InlineData("{A:=20, B:=30, C:=true}")]
+            [InlineData("{A:20+}")]
+            [InlineData("{A:20, B:30; }")]
+            [InlineData("{A:20, B:30 ++ }")]
+        public void TestParseRecordsNegative(string script)
+        {
+            TestParseErrors(script);
+        }
+
+        [Theory]
+        [InlineData("[]", "[  ]")]
+        [InlineData("[  ]")]
+        [InlineData("[ 1, 2, 3, 4, 5 ]")]
+        [InlineData("[ Abs(12), Abs(-12), Abs(45), Abs(-45), X!Y + Y!Z ]")]
+        [InlineData("[ Abs(12), Abs(-12), Abs(45), Abs(-45), X.Y + Y.Z ]")]
+        [InlineData("[ \"a\", \"b\", \"c\" ]")]
+
+        // Nested
+        [InlineData("[ [ 1, 2, 3 ], [ 4, 5, 6 ], [ \"a\", \"b\", \"c\" ] ]")]
+        public void TestParseTables(string script, string expected = null)
+        {
+            TestRoundtrip(script, expected);
+
+            // Nested
+            TestRoundtrip("[ [ 1, 2, 3 ], [ 4, 5, 6 ], [ \"a\", \"b\", \"c\" ] ]");
+        }
+
+        [Theory]
+        [InlineData("[a:10]")]
+        [InlineData("[a:10, b:20]")]
+        [InlineData("[10; 20; 30]")]
+        [InlineData("[10 20 30]")]
+        public void TestParseTables_Negative(string script)
+        {
+            TestParseErrors(script);
+        }
+
+        [Theory]
+        [InlineData("a=10;")]
+        [InlineData("a=b=10;")]
+        [InlineData("a=10;c=20;")]
+        public void TestFormulasParse(string script)
```

Comment:
>TestFormulasParse [](http://example.com/codeflow?start=20&length=17)

These are the new tests? Can you add the various examples from the table here?

Reply:
Added all the examples from our table.

Reply:
@gregli-msft  fyi